keras/callbacks.py

```diff
@@ -538,11 +541,13 @@ def on_epoch_end(self, epoch, logs=None):
         if self.monitor_op(current - self.min_delta, self.best):
             self.best = current
             self.wait = 0
+            self.good_weights = self.model.get_weights()
         else:
             self.wait += 1
             if self.wait >= self.patience:
                 self.stopped_epoch = epoch
                 self.model.stop_training = True
+                self.model.set_weights(self.good_weights)
```

Comment:
Please print a message for this action (like we do in `on_train_end`): "Restoring model to its state at the end of the best epoch."